src/main/java/de/tum/in/www1/artemis/web/rest/dto/competency/NgxLearningPathDTO.java

```diff
@@ -0,0 +1,43 @@
+package de.tum.in.www1.artemis.web.rest.dto.competency;
+
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+/**
+ * Represents simplified learning path optimized for Ngx representation
+ */
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public record NgxLearningPathDTO(Set<Node> nodes, Set<Edge> edges) {
+
+    public record Node(String id, NodeType type, Long linkedResource, Long linkedResourceParent, boolean completed, String label) {
+
+        public Node(String id, NodeType type, Long linkedResource, boolean completed, String label) {
```

Comment:
Unrelated to this PR but just as a reminder that we should discuss this.
@bassner what's your opinion on that? Can/should be disable this at least for classes/records in the dto package?